1.Perkenalan

Nama: Praduta Firman

Live Code 2

2. Judul

Promblems


1. Diketahui toko buah X menjual 10 produk dengan label nama A-J. Data penjualan serta pendapatan total toko X selalu dicatat setiap harinya dan berikut data penjualan 10 hari terakhir:


Hari ke-	A	B	C	D	E	F	G	H	I	J	Pendapatan
1	14	19	2	44	9	1	3	40	17	22	1402500
2	7	26	26	24	54	2	2	4	22	18	2325000
3	37	54	1	43	46	23	37	36	18	17	3307000
4	19	19	45	41	43	21	38	51	56	37	4077000
5	47	28	50	10	44	54	3	47	2	21	3998000
6	12	41	31	25	44	13	16	9	47	3	2674500
7	20	19	53	56	34	6	60	37	35	49	4014000
8	60	44	1	16	9	24	55	55	30	10	2754500
9	41	55	50	28	59	54	40	9	45	34	5511500
10	14	11	39	2	26	17	51	19	38	39	3338500st item

Berapa harga masing-masing produk A-J?




3. Import Pustaka


In [11]:
import pandas as pd
import numpy as np
import sympy as sy

In [14]:
hari1 = np.array([[14,19,2,44,9,1,3,40,17,22,1402500],
                  [7,26,26,24,54,2,2,4,22,18,2325000],
                  [37,54,1,43,46,23,37,36,18,17,3307000],
                  [19,19,45,41,43,21,38,51,56,37,4077000],
                  [47,28,50,10,44,54,3,47,2,21,3998000],
                  [12,41,31,25,44,13,16,9,47,3,2674500],
                  [20,19,53,56,34,6,60,37,35,49,4014000],
                  [60,44,1,16,9,24,55,55,30,10,2754500],
                  [41,55,50,28,59,54,40,9,45,34,5511500],
                  [14,11,39,2,26,17,51,19,38,39,3338500]])

In [84]:
a = np.array([[14,7,37,19,47,12,20,60,41,14]])
b = np.array([[19,26,54,19,28,41,19,44,55,11]])
c = np.array([[2,26,1,45,50,31,53,1,50,39]])
d = np.array([[44,24,43,41,10,25,56,16,28,2]])
e = np.array([[9,54,46,43,44,44,34,24,59,26]])
f = np.array([[1,2,23,21,54,13,6,24,54,17]])
g = np.array([[3,2,37,38,3,16,60,55,40,51]])
h = np.array([[40,4,36,51,47,9,37,55,9,19]])
i = np.array([[17,22,18,56,2,47,35,30,45,38]])
j = np.array([[22,18,17,37,21,3,49,10,34,39]])

a

array([[14,  7, 37, 19, 47, 12, 20, 60, 41, 14]])

In [67]:
b

array([[19, 26, 54, 19, 28, 41, 19, 44, 55, 11]])

In [68]:
c

array([[ 2, 26,  1, 45, 50, 31, 53,  1, 50, 39]])

In [69]:
d

array([[44, 24, 43, 41, 10, 25, 56, 16, 28,  2]])

In [70]:
e

array([[ 9, 54, 46, 43, 44, 44, 34, 24, 59, 26]])

In [71]:
f

array([[ 1,  2, 23, 21, 54, 13,  6, 24, 54, 17]])

In [72]:
g

array([[ 3,  2, 37, 38,  3, 16, 60, 55, 40, 51]])

In [73]:
h

array([[40,  4, 36, 51, 47,  9, 37, 55,  9, 19]])

In [74]:
i

array([[17, 22, 18, 56,  2, 47, 35, 30, 45, 38]])

In [75]:
j

array([[22, 18, 17, 37, 21,  3, 49, 10, 34, 39]])

In [76]:
a
b
c
d
e
f
g
h
i
j
unique, counts = np.unique(a, return_counts=True)

In [77]:
  OBE = sy.Matrix(a)
  OBE.rref()

(Matrix([[1, 1/2, 37/14, 19/14, 47/14, 6/7, 10/7, 30/7, 41/14, 1]]), (0,))

In [78]:
a.T

array([[14],
       [ 7],
       [37],
       [19],
       [47],
       [12],
       [20],
       [60],
       [41],
       [14]])

In [106]:
a.sum()

257

In [107]:
b.sum()

316

In [79]:
b = np.array([[19,26,54,19,28,41,19,44,55,11]])
ab = sy.Matrix(b)
ab.rref()

(Matrix([[1, 26/19, 54/19, 1, 28/19, 41/19, 1, 44/19, 55/19, 11/19]]), (0,))

In [80]:
np.linalg.norm(a)

101.01980003939822

4. Data Loading

Soal 1 :

In [81]:
a = np.array(counts)
a

array([1, 1, 2, 1, 1, 1, 1, 1, 1])

In [91]:

a=np.array([[14,7,37],
          [19,47,12],
          [20,60,41]])
eigVal,eigVec=np.linalg.eig(a)


2. Menggunakan konsep integral, tentukan luas negara Swiss jika diketahui koordinat batas-batasnya! Dataset koordinat batas Swiss dapat diakses pada link berikut: https://raw.githubusercontent.com/fahmimnalfrzki/Dataset/main/Swiss_boundaries.txt?token=AEZDEHXBTS7OMF32RTYPKZDA2FELA

Soal 2 :

In [124]:
df = pd.read_csv('https://raw.githubusercontent.com/fahmimnalfrzki/Dataset/main/Swiss_boundaries.txt?token=AEZDEHXBTS7OMF32RTYPKZDA2FELA')
df

longitude  latitude
0      9.56672  47.54045
1      9.55980  47.50209
2      9.59660  47.46944
3      9.64965  47.45861
4      9.67035  47.39069
..         ...       ...
273    9.03333  47.68833
274    9.08264  47.68541
275    9.26110  47.66284
276    9.47832  47.57585
277    9.56672  47.54045

[278 rows x 2 columns]

In [125]:
df.head()

longitude  latitude
0    9.56672  47.54045
1    9.55980  47.50209
2    9.59660  47.46944
3    9.64965  47.45861
4    9.67035  47.39069

In [126]:
len(df)

278

In [127]:
df.shape

(278, 2)

In [128]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 278 entries, 0 to 277
Data columns (total 2 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   longitude  278 non-null    float64
 1   latitude   278 non-null    float64
dtypes: float64(2)
memory usage: 4.5 KB


In [129]:
df.describe()

longitude    latitude
count  278.000000  278.000000
mean     8.388293   46.789161
std      1.308479    0.627381
min      5.967010   45.829440
25%      7.339238   46.251680
50%      8.572925   46.616660
75%      9.500760   47.460065
max     10.488210   47.806660

In [150]:
center = (df.min() + df.max())/2
center

longitude     8.22761
latitude     46.81805
dtype: float64

In [151]:
nortlong = []
for i in df.longitude:
  if i >= 8.22761:
    nortlong.append(i)
nortlong

[9.566720000000089,
 9.559800000000052,
 9.59660000000008,
 9.649650000000065,
 9.670350000000044,
 9.637360000000058,
 9.610970000000068,
 9.540000000000076,
 9.533570000000054,
 9.496800000000064,
 9.48500000000007,
 9.487080000000047,
 9.502080000000092,
 9.511730000000057,
 9.474640000000079,
 9.485280000000046,
 9.506390000000067,
 9.541390000000092,
 9.598630000000073,
 9.632500000000052,
 9.719440000000077,
 9.87292000000008,
 9.89160000000004,
 9.875830000000065,
 9.877780000000088,
 9.935000000000061,
 10.058890000000076,
 10.109440000000065,
 10.131530000000057,
 10.14181000000008,
 10.16083000000009,
 10.227500000000076,
 10.313330000000065,
 10.323330000000057,
 10.350620000000047,
 10.390760000000057,
 10.430000000000064,
 10.488210000000038,
 10.47124000000008,
 10.447500000000046,
 10.470830000000092,
 10.486530000000073,
 10.485690000000089,
 10.471870000000079,
 10.457530000000077,
 10.44972000000007,
 10.33722000000006,
 10.303610000000047,
 10.246320000000082,
 10.25

In [152]:
southlong = []
for i in df.longitude:
  if i <= 8.22761:
    southlong.append(i)
southlong

[8.141940000000089,
 8.158750000000055,
 8.149580000000071,
 8.009030000000052,
 7.9730500000000575,
 7.950000000000045,
 7.906810000000063,
 7.8793000000000575,
 7.866390000000081,
 7.855740000000082,
 7.783890000000043,
 7.743330000000071,
 7.709170000000086,
 7.650280000000067,
 7.536670000000073,
 7.458060000000047,
 7.429720000000088,
 7.386110000000088,
 7.338890000000049,
 7.29681000000005,
 7.23028000000005,
 7.191390000000069,
 7.10326000000009,
 7.070280000000082,
 7.038050000000055,
 7.030550000000062,
 7.007220000000075,
 6.933050000000037,
 6.873890000000074,
 6.7834700000000785,
 6.800000000000068,
 6.80702000000008,
 6.798330000000077,
 6.7377800000000425,
 6.702780000000075,
 6.63361000000009,
 6.5151400000000885,
 6.491940000000057,
 6.319720000000075,
 6.29556000000008,
 6.244860000000074,
 6.230000000000075,
 6.243750000000091,
 6.271950000000062,
 6.245550000000037,
 6.187500000000057,
 6.133260000000063,
 5.996390000000076,
 5.967010000000073,
 5.981390000000089,
 

In [154]:
nortlat = []
for i in df.latitude:
  if i >= 46.81805:
    nortlat.append(i)
nortlat

[47.54045000000008,
 47.50209000000007,
 47.46944000000008,
 47.45861000000008,
 47.39069000000006,
 47.369580000000035,
 47.362220000000036,
 47.284160000000035,
 47.274540000000066,
 47.223330000000026,
 47.196240000000046,
 47.172630000000076,
 47.146810000000066,
 47.10597000000007,
 47.05746000000005,
 47.06110000000007,
 47.06472000000008,
 47.07000000000005,
 47.06384000000008,
 47.05667000000005,
 47.050280000000036,
 47.02305000000007,
 47.00034000000005,
 46.968180000000075,
 46.940410000000035,
 46.91277000000007,
 46.86638000000005,
 46.85027000000007,
 46.85041000000007,
 46.85134000000005,
 46.854990000000036,
 46.87694000000005,
 46.933610000000044,
 46.955550000000066,
 46.99124000000007,
 47.00257000000005,
 46.984160000000095,
 46.93599000000007,
 46.871350000000064,
 46.839440000000074,
 46.895830000000046,
 46.971670000000074,
 46.980000000000075,
 47.000000000000064,
 47.034160000000035,
 47.063890000000065,
 47.13083000000007,
 47.164990000000046,
 47.171520000000

In [155]:
soutlat = []
for i in df.latitude:
  if i <= 46.81805:
    soutlat.append(i)
soutlat

[46.76305000000008,
 46.63583000000007,
 46.617350000000044,
 46.592640000000074,
 46.55576000000008,
 46.54251000000005,
 46.53916000000004,
 46.54944000000007,
 46.554860000000076,
 46.59041000000008,
 46.61597000000007,
 46.63444000000004,
 46.632220000000075,
 46.61138000000005,
 46.54152000000005,
 46.502640000000035,
 46.44559000000004,
 46.42375000000004,
 46.39027000000005,
 46.263120000000065,
 46.23069000000004,
 46.221800000000094,
 46.29167000000007,
 46.333050000000064,
 46.355200000000075,
 46.379580000000026,
 46.38319000000007,
 46.36972000000009,
 46.35833000000007,
 46.34221000000008,
 46.30027000000007,
 46.29194000000007,
 46.29694000000006,
 46.30624000000007,
 46.319720000000075,
 46.386240000000036,
 46.426670000000065,
 46.472500000000075,
 46.508050000000075,
 46.500830000000065,
 46.44242000000008,
 46.425760000000075,
 46.350970000000075,
 46.324860000000065,
 46.292500000000075,
 46.23363000000007,
 46.17861000000005,
 46.16805000000005,
 46.160270000000075,

In [168]:
luas = df.longitude * df.latitude
luas

0      454.806174
1      454.110480
2      455.545228
3      457.958976
4      458.284559
          ...    
273    430.784422
274    433.109412
275    441.410328
276    450.939131
277    454.806174
Length: 278, dtype: float64

5. Mathematical Calculation

Soal 1 :

In [92]:
eigVal

array([86.66282951 +0.j        ,  7.66858524+13.34279616j,
        7.66858524-13.34279616j])

In [ ]:

eigVec

array([[ 0.43980992+0.j        ,  0.85319479+0.j        ,
         0.85319479-0.j        ],
       [ 0.4464514 +0.j        , -0.31361611-0.21253055j,
        -0.31361611+0.21253055j],
       [ 0.77926144+0.j        , -0.08666533+0.34788427j,
        -0.08666533-0.34788427j]])

In [93]:
np.fliplr(np.diag(eigVal))

array([[ 0.         +0.j        ,  0.         +0.j        ,
        86.66282951 +0.j        ],
       [ 0.         +0.j        ,  7.66858524+13.34279616j,
         0.         +0.j        ],
       [ 7.66858524-13.34279616j,  0.         +0.j        ,
         0.         +0.j        ]])

Soal 2 :

In [169]:
luas = df.longitude * df.latitude
luas

0      454.806174
1      454.110480
2      455.545228
3      457.958976
4      458.284559
          ...    
273    430.784422
274    433.109412
275    441.410328
276    450.939131
277    454.806174
Length: 278, dtype: float64

6. Hasil

1. Jadi hasil masing - masing produk A-J adalah 100.0198

2. Jadi luas negara Swiss adalah  454.806174 m2